docusaurus/docs/tutorials/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,235 @@
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // The `jsonData` property is an object that your plugin can manage
+    // using the Grafana Rest APIs
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+### Access plugin state in a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+import { MyCustomDataProvider } from './MyCustomDataProvider';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => (
+    <MyCustomDataProvider>
+      <div>Plugin specific setting: {meta.jsonData.foo}</div>
+    </MyCustomDataProvider>
+  ),
+});
+```
+
+### Hide a component in certain conditions
+
+This is nothing new for developers working with React, just return `null` from your component in order to not render anything (to hide it).
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // For the sake of the example this condition is relying on a `jsonData` property
+    // that is managed by your plugin
+    if (!meta.jsonData.isExtensionEnabled) {
+      return null;
+    }
+
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+## Troubleshooting
+
+### My link is not appearing
+
+1. **Check the console logs** - there is a good chance that your link is not appearing due to some validation errors. In this case you should see some relevant logs in your browsers console.
+1. **Check the `targets`** - make sure that you are using the correct extension point ids (always use the `PluginExtensionPoints` enum for Grafana extension points)
+1. **Check the links `configure()` function** - in case your link has a `configure()` function it can happen that it is returning `undefined` under certain conditions, which hides the link.
+1. **Check if you register too many links** - certain extension points limit the number of links allowed per plugin, and in case your plugin registers more than one links for the same extension point there is a chance that some of them are filtered out.
+1. **Check the Grafana version** - link and component extensions are only supported after Grafana version **`>=10.1.0`**, while `.addLink()` is only supported in versions **>=`11.1.0`**.
+
+### My component is not appearing
```

Comment:
there is a lot of overlap, could we combine most into My extension is not appearing?

Reply:
Yeah, I think they could be combined indeed. Maybe I would leave that for an upcoming PR though - just to make this one easier to manage.

Reply:
Added an item to the epic to combine these two into one section called "My extension is not appearing?"

https://github.com/grafana/grafana-community-team/issues/131